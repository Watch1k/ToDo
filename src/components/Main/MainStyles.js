export const layoutStyles = {
  padding: 30,
  border: '1px solid #ddd',
  background: '#e6f7ff'
}
