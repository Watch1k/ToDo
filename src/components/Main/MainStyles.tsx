import React from 'react'

export const layoutMainStyles: React.CSSProperties = {
  padding: 30,
  border: '1px solid #ddd',
  background: '#e6f7ff',
}
