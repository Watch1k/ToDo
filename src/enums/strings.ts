export enum Strings {
  header = 'ToDo List',
  darkMode = 'Dark Mode',
  mainCreateTusk = 'Create task',
  mainListOfTasks = 'List of tasks',
  results = 'Results',
  search = 'Search',
}
