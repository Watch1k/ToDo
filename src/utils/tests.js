import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'

export const renderWithRedux = (ui, store) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store,
})
