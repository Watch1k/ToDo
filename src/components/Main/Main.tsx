import React from 'react'
import { Divider, Layout, Row } from 'antd'
import { layoutStyles } from './MainStyles'
import FormContainer from '../../containers/FormContainer'
import ListContainer from '../../containers/ListContainer'
import DarkMode from '../DarkMode'
import Filter from '../Filter'
import { Strings } from '../../enums/strings'

export const Main = () => (
  <Layout style={layoutStyles} data-testid='main'>
    <Divider>{Strings.mainCreateTusk}</Divider>
    <Row type='flex' justify='center'>
      <FormContainer />
    </Row>
    <Row type='flex' justify='center' style={{ margin: '30px 0' }}>
      <Filter />
    </Row>

    <Divider>{Strings.mainListOfTasks}</Divider>
    <Row type='flex' justify='center'>
      <ListContainer />
    </Row>

    <Divider>{Strings.darkMode}</Divider>
    <Row type='flex' justify='center' style={{ margin: '30px 0' }}>
      <DarkMode />
    </Row>
  </Layout>
)

export default Main
