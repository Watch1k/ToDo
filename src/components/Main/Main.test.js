import React from 'react'
import Main from './Main'
import { Strings } from '../../enums/strings'
import { renderWithRedux } from '../../utils/tests'
import { store } from '../../store'

describe('Main component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = renderWithRedux(<Main />, store)
  })

  it('Renders', () => {
    expect(wrapper.getByTestId('main')).toBeInTheDocument()
  })

  it('Matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('All text lines are present', () => {
    ;[Strings.mainListOfTasks, Strings.mainCreateTusk].forEach(text =>
      expect(wrapper.getByText(text)).toBeInTheDocument()
    )
  })
})
