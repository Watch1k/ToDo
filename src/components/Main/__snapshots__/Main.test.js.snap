// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="ant-layout"
    data-testid="main"
    style="padding: 30px; border: 1px solid #ddd; background: rgb(230, 247, 255);"
  >
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text-center"
      role="separator"
    >
      <span
        class="ant-divider-inner-text"
      >
        Create task
      </span>
    </div>
    <div
      class="ant-row-flex ant-row-flex-center"
    >
      <form
        class="ant-form ant-form-inline"
        data-testid="form"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <span
                  class="ant-input-affix-wrapper"
                >
                  <span
                    class="ant-input-prefix"
                  >
                    <i
                      aria-label="icon: unordered-list"
                      class="anticon anticon-unordered-list"
                      style="color: rgba(0, 0, 0, 0.25);"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="unordered-list"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M912 192H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM104 228a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm0 284a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm0 284a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
                        />
                      </svg>
                    </i>
                  </span>
                  <input
                    class="ant-input"
                    data-testid="form-input"
                    placeholder="New task..."
                    type="text"
                    value=""
                  />
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control"
            >
              <span
                class="ant-form-item-children"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="submit"
                >
                  <span>
                    Create
                  </span>
                </button>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="ant-row-flex ant-row-flex-center"
      style="margin: 30px 0px;"
    >
      <div
        class="ant-card ant-card-bordered"
        style="text-align: center;"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Filter:
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-radio-group ant-radio-group-outline"
          >
            <label
              class="ant-radio-wrapper ant-radio-wrapper-checked"
            >
              <span
                class="ant-radio ant-radio-checked"
              >
                <input
                  checked=""
                  class="ant-radio-input"
                  type="radio"
                  value="SHOW_ACTIVE"
                />
                <span
                  class="ant-radio-inner"
                />
              </span>
              <span>
                Show active
              </span>
            </label>
            <label
              class="ant-radio-wrapper"
            >
              <span
                class="ant-radio"
              >
                <input
                  class="ant-radio-input"
                  type="radio"
                  value="SHOW_COMPLETED"
                />
                <span
                  class="ant-radio-inner"
                />
              </span>
              <span>
                Show completed
              </span>
            </label>
            <label
              class="ant-radio-wrapper"
            >
              <span
                class="ant-radio"
              >
                <input
                  class="ant-radio-input"
                  type="radio"
                  value="SHOW_ALL"
                />
                <span
                  class="ant-radio-inner"
                />
              </span>
              <span>
                Show all
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text-center"
      role="separator"
    >
      <span
        class="ant-divider-inner-text"
      >
        List of tasks
      </span>
    </div>
    <div
      class="ant-row-flex ant-row-flex-center"
    >
      <div
        class="ant-list ant-list-split"
        data-testid="list"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              style="text-align: center;"
            >
              No data
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-with-text-center"
      role="separator"
    >
      <span
        class="ant-divider-inner-text"
      >
        Dark Mode
      </span>
    </div>
    <div
      class="ant-row-flex ant-row-flex-center"
      style="margin: 30px 0px;"
    >
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <button
          aria-checked="true"
          class="ant-switch ant-switch-checked"
          data-testid="darkMode"
          role="switch"
          type="button"
        >
          <span
            class="ant-switch-inner"
          />
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
